htslib/hts.h

```diff
@@ -366,6 +366,7 @@ extern "C" {
 
     int hts_idx_get_stat(const hts_idx_t* idx, int tid, uint64_t* mapped, uint64_t* unmapped);
     uint64_t hts_idx_get_n_no_coor(const hts_idx_t* idx);
+    char *hts_idx_getfn(const char *fn, const char *ext);
```

Comment:
On Fri, Feb 13, 2015 at 03:47:25AM -0800, John Marshall wrote:
> > @@ -366,6 +366,7 @@ extern "C" {
> >  
> >      int hts_idx_get_stat(const hts_idx_t* idx, int tid, uint64_t* mapped, uint64_t* unmapped);
> >      uint64_t hts_idx_get_n_no_coor(const hts_idx_t* idx);
> > +    char *hts_idx_getfn(const char *fn, const char *ext);
> 
> This is an htslib-internal function -- it has weird side-effects due to calling `test_and_fetch()`.  So I think we need to come up with a better solution than declaring it in a public header...

I'm not really happy with the alternative solution of moving all the
cram index code into hts.c either.  I was sort of assuming all of the
hts_idx_* functions weren't really designed to be properly external
anyway as none of them have the usual comments describing the
functions that most (well some!) of our public API does.

Maybe we need an hts_internal.h or some such?

James

-- 
James Bonfield (jkb@sanger.ac.uk) | Hora aderat briligi. Nunc et Slythia Tova
                                  | Plurima gyrabant gymbolitare vabo;
  A Staden Package developer:     | Et Borogovorum mimzebant undique formae,
https://sf.net/projects/staden/   | Momiferique omnes exgrabure Rathi. 


-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.